src/Hamiltonians/ReducedDensityMatrix.jl

```diff
@@ -0,0 +1,150 @@
+"""
+    SingleParticleReducedDensityMatrix(i, j) <: AbstractHamiltonian
+
+Represent a {i,j} element of the single-particle reduced density matrix:
+
+```math
+\\hat{ρ}^{(1)}_{i,j} = \\langle \\psi | \\hat a^†_{i} \\hat a_{j} | \\psi \\rangle
+```
+
+where ``i`` and ``j`` are the `mode` and ``| \\psi \\rangle`` is the `state-ket` of the given Hamiltonian.
+
+# See also
+
+* [`single_particle_density`](@ref)
+* [`SingleParticleDensity`](@ref)
+* [`TwoParticleReducedDensityMatrix`](@ref)
+"""
+struct SingleParticleReducedDensityMatrix <: AbstractHamiltonian{Float64}
+    I::Int
+    J::Int
+end
+
+function Base.show(io::IO, spd::SingleParticleReducedDensityMatrix)
+    print(io, "SingleParticleReducedDensityMatrix($(spd.I), $(spd.J))")
+end
+
+LOStructure(::Type{T}) where T<:SingleParticleReducedDensityMatrix = AdjointUnknown()
+
+function diagonal_element(spd::SingleParticleReducedDensityMatrix, add::SingleComponentFockAddress)
+    src = find_mode(add, spd.J)
+    dst = find_mode(add,spd.I)
+    address, value = excitation(add, (dst,), (src,))
+    if spd.I == spd.J
+        return value
+    else
+        return 0.0
+    end
+end
+
+function num_offdiagonals(spd::SingleParticleReducedDensityMatrix, address::SingleComponentFockAddress)
+    if spd.I == spd.J
+        return 0
+    else
+        return 2
+    end
+end
+
+function get_offdiagonal(spd::SingleParticleReducedDensityMatrix, add::SingleComponentFockAddress, chosen)
+    src = find_mode(add, spd.J)
+    dst = find_mode(add,spd.I)
+    if chosen == 2
+        src, dst = dst, src
+    end
```

Comment:
The idea here is that this shouldn't change the result, but by applying both we're increasing the probability to get a hit when computing the overlap between two sampled vectors. It would be good to check how much it actually reduces the noise though.